biz.aQute.bndlib.tests/test/aQute/bnd/build/classindex/ClassIndexTest.java

```diff
@@ -20,50 +18,97 @@
 import aQute.libg.map.MAP.MAPX;
 
 public class ClassIndexTest {
-	public static final String	TMPDIR		= "generated/tmp/test";
+	public static final String			TMPDIR	= "generated/tmp/test";
+	private static File					testDir;
+	private static File					home	= IO.getFile("testresources/classindex");
+	private static Workspace			ws;
+
 	@Rule
-	public final TestName		testName	= new TestName();
-	private File				testDir;
-	private File				home		= IO.getFile("testresources/classindex");
+	public final JUnitSoftAssertions	softly	= new JUnitSoftAssertions();
 
-	@Before
-	public void setUp() throws IOException {
-		testDir = new File(TMPDIR, getClass().getName() + "/" + testName.getMethodName());
+	@BeforeClass
+	public static void setUp() throws Exception {
```

Comment:
Each test method should have its own tmp dir so there is no chance of side effects between tests. This should be `@BeforeEach`.

Reply:
Setting up the workspace afresh for each test comes at performance penalty. I believe that this would be necessary if the tests were attempting to modify the workspace in some way, but as the workspace is read-only in these tests it seems like overkill to me.

Reply:
Lets take the perf hit. I want to test cases to be side effect free and that is how we guarantee that. We do this in many other test cases. Also, a workspace is not read-only. Many things can be going on since there is a lot of code involved.

Reply:
Done.